pkg/config/derived-cert-subject.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 LY Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"crypto/x509/pkix"
+	"fmt"
+
+	"slices"
+)
+
+type DerivedCertSubject struct {
+	roleCert    pkix.Name // private as the derived state is used only within the config package
+	serviceCert pkix.Name // private as the derived state is used only within the config package
+}
+
+// derivedCertSubject sets the DerivedCertSubject with the given rawCertSubject.
+// 1. Parses the string in RFC 4514 format to pkix.Name struct.
+// 2. Returns error if CN, SERIALNUMBER and other unknown attributes are set.
+// 3. Apply default value if the corresponding attribute is not set.
+func (idCfg *IdentityConfig) derivedCertSubject() error {
```

Comment:
note: early return is not possible because we have the default variables like `DEFAULT_COUNTRY`